File: frontend/src/api/esgMonitoring.ts
import { apiRequest } from "./client";

export interface EsgMonitoringKpi {
  totalSuppliers: number;
  highEsgRiskSuppliers: number;
  deterioratingSuppliers: number;
  openEsgAlerts: number;
  averageEsgHealth: number;
}

export interface EsgIndicatorSummary {
  key: string;
  label: string;
  category: string;
  averageRisk: number;
  highRiskSuppliers: number;
  trend: string;
  description: string;
  isPriority: boolean;
}

export interface EsgWatchlistSupplier {
  supplierId: number;
  supplierName: string;
  country: string | null;
  tier: string | null;
  esgRiskScore: number;
  esgHealthScore: number;
  bwsRisk: number;
  hrrRisk: number;
  landUseRisk: number;
  trend: string;
  status: string;
  primaryConcern: string;
  recommendedAction: string;
}

export interface EsgAlertItem {
  id: string;
  supplierId: number;
  supplierName: string;
  severity: string;
  indicator: string;
  message: string;
  recommendedAction: string;
}

export interface EsgHealthTrend {
  label: string;
  supplierCount: number;
}

export interface EsgMlFlaggedSupplier {
  supplierId: number;
  supplierName: string;
  anomalyScore: number;
  confidence: number;
  signal: string;
}

export interface EsgMlInsights {
  modelName: string;
  monitoringMode: string;
  flaggedSuppliers: number;
  averageAnomalyScore: number;
  topSignals: string[];
  dataLimitations: string[];
  flaggedSupplierDetails: EsgMlFlaggedSupplier[];
}

export interface EsgMonitoringOverview {
  kpis: EsgMonitoringKpi;
  indicators: EsgIndicatorSummary[];
  watchlist: EsgWatchlistSupplier[];
  alerts: EsgAlertItem[];
  healthTrends: EsgHealthTrend[];
  mlInsights: EsgMlInsights;
}

export async function getEsgMonitoringOverview(): Promise<EsgMonitoringOverview> {
  return apiRequest<EsgMonitoringOverview>("/esg-monitoring/overview");
}

File: backend/app/schemas/esg_monitoring.py
from pydantic import BaseModel, ConfigDict


class EsgMonitoringKpi(BaseModel):
    totalSuppliers: int
    highEsgRiskSuppliers: int
    deterioratingSuppliers: int
    openEsgAlerts: int
    averageEsgHealth: float

    model_config = ConfigDict(from_attributes=True)


class EsgIndicatorSummary(BaseModel):
    key: str
    label: str
    category: str
    averageRisk: float
    highRiskSuppliers: int
    trend: str
    description: str
    isPriority: bool = False

    model_config = ConfigDict(from_attributes=True)


class EsgWatchlistSupplier(BaseModel):
    supplierId: int
    supplierName: str
    country: str | None = None
    tier: str | None = None
    esgRiskScore: float
    esgHealthScore: float
    bwsRisk: float
    hrrRisk: float
    landUseRisk: float
    mlAnomalyScore: float
    mlConfidence: float
    trend: str
    status: str
    primaryConcern: str
    recommendedAction: str

    model_config = ConfigDict(from_attributes=True)


class EsgAlertItem(BaseModel):
    id: str
    supplierId: int
    supplierName: str
    severity: str
    indicator: str
    message: str
    recommendedAction: str

    model_config = ConfigDict(from_attributes=True)


class EsgHealthTrend(BaseModel):
    label: str
    supplierCount: int

    model_config = ConfigDict(from_attributes=True)


class EsgMlFlaggedSupplier(BaseModel):
    supplierId: int
    supplierName: str
    anomalyScore: float
    confidence: float
    signal: str

    model_config = ConfigDict(from_attributes=True)


class EsgMlInsights(BaseModel):
    modelName: str
    monitoringMode: str
    flaggedSuppliers: int
    averageAnomalyScore: float
    topSignals: list[str]
    dataLimitations: list[str]
    flaggedSupplierDetails: list[EsgMlFlaggedSupplier]

    model_config = ConfigDict(from_attributes=True)


class EsgMonitoringOverview(BaseModel):
    kpis: EsgMonitoringKpi
    indicators: list[EsgIndicatorSummary]
    watchlist: list[EsgWatchlistSupplier]
    alerts: list[EsgAlertItem]
    healthTrends: list[EsgHealthTrend]
    mlInsights: EsgMlInsights

    model_config = ConfigDict(from_attributes=True)
